<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>628</width>
    <height>413</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>2</horstretch>
    <verstretch>2</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Serial HEX Console</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#MainWindow {background-color: rgb(255, 255, 255);}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" rowspan="4">
     <widget class="QWidget" name="settingsToolbar" native="true">
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">#settingsToolbar { border-style: solid; border-width: 1px; border-color: rgb(0, 0, 0); }

</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="settings_label">
         <property name="font">
          <font>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string notr="true">border-width: 0px;
border-style: solid;
border-color: rgb(0, 0, 0);</string>
         </property>
         <property name="text">
          <string>Settings</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="Line" name="line">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true"> border-style: solid; border-width: 1px; border-color: rgb(0, 0, 0);</string>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="script_Label">
         <property name="font">
          <font>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">border-width: 0px;
border-style: solid;
border-color: rgb(0, 0, 0);</string>
         </property>
         <property name="text">
          <string>Script</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <layout class="QFormLayout" name="settings_layout">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="comPort_label">
           <property name="styleSheet">
            <string notr="true">border-width: 0px;</string>
           </property>
           <property name="text">
            <string>Com port</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comPort_comboBox">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);
border-bottom-style: solid;
 border-bottom-width: 1px; 
border-bottom-color: rgb(0, 0, 0); </string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="bitsRate_label">
           <property name="styleSheet">
            <string notr="true">border-width: 0px;</string>
           </property>
           <property name="text">
            <string>Bits rate</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="bitsRate_comboBox">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="dataBits_label">
           <property name="styleSheet">
            <string notr="true">border-width: 0px;</string>
           </property>
           <property name="text">
            <string>Data bits</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="dataBits_comboBox">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="parity_label">
           <property name="styleSheet">
            <string notr="true">border-width: 0px;</string>
           </property>
           <property name="text">
            <string>Parity</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="parity__comboBox">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="stopBits_label">
           <property name="styleSheet">
            <string notr="true">border-width: 0px;</string>
           </property>
           <property name="text">
            <string>Stop bits</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="stopBits_comboBox">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="flowControl_label">
           <property name="styleSheet">
            <string notr="true">border-width: 0px;</string>
           </property>
           <property name="text">
            <string>Flow control</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QComboBox" name="flowControl_comboBox">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="5" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="run_button">
           <property name="text">
            <string>Run</string>
           </property>
           <property name="shortcut">
            <string>Ctrl+R</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="4" column="0">
        <layout class="QFormLayout" name="script_layout">
         <item row="0" column="0">
          <widget class="QPushButton" name="open_button">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <stylestrategy>PreferAntialias</stylestrategy>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="text">
            <string>Open</string>
           </property>
           <property name="shortcut">
            <string>Ctrl+O</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
           <property name="default">
            <bool>false</bool>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="filePath_lineEdit">
           <property name="styleSheet">
            <string notr="true">border-style: solid; 
border-width: 1px; 
border-color: rgb(0, 0, 0); </string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
      <zorder>settings_label</zorder>
      <zorder>script_Label</zorder>
      <zorder>line</zorder>
      <zorder></zorder>
      <zorder></zorder>
     </widget>
    </item>
    <item row="1" column="1" rowspan="2">
     <widget class="QPushButton" name="clear_button">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>Clear</string>
      </property>
      <property name="shortcut">
       <string>Ctrl+Backspace</string>
      </property>
      <property name="autoDefault">
       <bool>false</bool>
      </property>
      <property name="default">
       <bool>false</bool>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>115</width>
        <height>25</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="3" rowspan="2">
     <layout class="QHBoxLayout" name="mode_horizontal_layout">
      <item>
       <widget class="QLabel" name="mode_label">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Mode</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true"> border-style: solid; border-width: 0px; border-color: rgb(0, 0, 0);</string>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QRadioButton" name="hex_radioButton">
           <property name="text">
            <string>HEX</string>
           </property>
           <property name="shortcut">
            <string>Ctrl+H</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="text_radioButton">
           <property name="text">
            <string>TEXT</string>
           </property>
           <property name="shortcut">
            <string>Ctrl+T</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="4">
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>114</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="2">
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>115</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="4">
     <spacer name="horizontalSpacer_4">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>114</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="1" colspan="4">
     <widget class="QTextBrowser" name="log_textBrowser">
      <property name="styleSheet">
       <string notr="true"> border-style: solid; border-width: 1px; border-color: rgb(0, 0, 0);</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1" colspan="4">
     <layout class="QGridLayout" name="sendEdit_layout">
      <item row="0" column="1">
       <widget class="QPushButton" name="send_pushButton">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>Send</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+Return</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLineEdit" name="send_text_lineEdit">
        <property name="styleSheet">
         <string notr="true">border-style: solid; 
border-width: 1px; 
border-color: rgb(0, 0, 0); </string>
        </property>
        <property name="maxLength">
         <number>128</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>clear_button</sender>
   <signal>clicked()</signal>
   <receiver>log_textBrowser</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>339</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>264</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
